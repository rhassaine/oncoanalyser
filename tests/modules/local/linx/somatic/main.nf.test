nextflow_process {
    name "Test Process LINX_SOMATIC"
    script "modules/local/linx/somatic/main.nf"
    process "LINX_SOMATIC"
    config "tests/nextflow.config"

    test("LINX_SOMATIC with default parameters") {
        when {
            params {
                test_sample_data = 'https://raw.githubusercontent.com/nf-core/test-datasets/oncoanalyser/sample_data/simulated_reads/wgts/'
                resource_path = '/home/rhassaine/resources_files_oa/38/'
                mode = 'WGS'
            }
            process {
                """
                input[0] = [
                    [id: 'test', sample_id: 'subject_a.tumor'],
                    file('/home/rhassaine/container_test_output/subject_a/purple/')
                ]
                input[1] = 38
                input[2] = file(params.resource_path + "hmf_pipeline_resources.38_v6.0--2/common/ensembl_data/", checkIfExists: true)
                input[3] = file(params.resource_path + "hmf_pipeline_resources.38_v6.0--2/dna/sv/known_fusion_data.38.csv", checkIfExists: true)
                input[4] = file(params.resource_path + "hmf_pipeline_resources.38_v6.0--2/common/DriverGenePanel.38.tsv", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.annotation_dir != null },
                { assert process.out.annotation_dir.size() > 0 },
                { assert snapshot(process.out).match() }
            )

            if (process.success) {
                def linx_output_dir = process.out.annotation_dir[0][1]
                def sample_id = process.out.annotation_dir[0][0].sample_id
                assertAll(
                    { assert path(linx_output_dir).resolve("linx.version").exists() },
                    { assert path(linx_output_dir).resolve("${sample_id}.linx.breakend.tsv").exists() },
                    { assert path(linx_output_dir).resolve("${sample_id}.linx.clusters.tsv").exists() },
                    { assert path(linx_output_dir).resolve("${sample_id}.linx.driver.catalog.tsv").exists() },
                    { assert path(linx_output_dir).resolve("${sample_id}.linx.drivers.tsv").exists() },
                    { assert path(linx_output_dir).resolve("${sample_id}.linx.fusion.tsv").exists() },
                    { assert path(linx_output_dir).resolve("${sample_id}.linx.links.tsv").exists() },
                    { assert path(linx_output_dir).resolve("${sample_id}.linx.neoepitope.tsv").exists() },
                    { assert path(linx_output_dir).resolve("${sample_id}.linx.svs.tsv").exists() },
                    { assert path(linx_output_dir).resolve("${sample_id}.linx.vis_copy_number.tsv").exists() },
                    { assert path(linx_output_dir).resolve("${sample_id}.linx.vis_fusion.tsv").exists() },
                    { assert path(linx_output_dir).resolve("${sample_id}.linx.vis_gene_exon.tsv").exists() },
                    { assert path(linx_output_dir).resolve("${sample_id}.linx.vis_protein_domain.tsv").exists() },
                    { assert path(linx_output_dir).resolve("${sample_id}.linx.vis_segments.tsv").exists() },
                    { assert path(linx_output_dir).resolve("${sample_id}.linx.vis_sv_data.tsv").exists() },
                    { assert process.out.versions != null },
                    { assert process.out.versions.size() > 0 },
                    { assert path(process.out.versions[0]).exists() }
                )
            }
        }
    }

    test("LINX_SOMATIC should fail when input is missing") {
        when {
            params {
                test_sample_data = 'https://raw.githubusercontent.com/nf-core/test-datasets/oncoanalyser/sample_data/simulated_reads/wgts/'
                resource_path = '/home/rhassaine/resources_files_oa/38/'
                mode = 'WGS'
            }
            process {
                """
                input[0] = []  // Missing input
                """
            }
        }

        then {
            assertAll(
                { assert !process.success },
                { assert process.exitStatus != 0 }
            )
        }
    }
}