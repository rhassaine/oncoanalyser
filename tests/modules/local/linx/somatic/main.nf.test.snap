{
    "LINX_SOMATIC with default parameters": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "sample_id": "subject_a.tumor"
                        },
                        [
                            "linx.version:md5,a5fc97b5c4e6092096f0a1fdffd20159",
                            "subject_a.tumor.linx.breakend.tsv:md5,78e4cce3e07d1b0016d0a71b49db0e57",
                            "subject_a.tumor.linx.clusters.tsv:md5,49209060d580237991391282dbbea3af",
                            "subject_a.tumor.linx.driver.catalog.tsv:md5,1b18c47117b810430c3197f84c1db659",
                            "subject_a.tumor.linx.drivers.tsv:md5,8fb6e60a850979da67e3d3801c4a56d7",
                            "subject_a.tumor.linx.fusion.tsv:md5,cfaab2c97b2bddc258d2a5e866c55c97",
                            "subject_a.tumor.linx.links.tsv:md5,b1d1360fa6598420a8d9751ba2a4d4dc",
                            "subject_a.tumor.linx.neoepitope.tsv:md5,3c76a4e3521ccd5ac627857c2c5d17de",
                            "subject_a.tumor.linx.svs.tsv:md5,d6f1abd6f8678858e28eafe43fad4964",
                            "subject_a.tumor.linx.vis_copy_number.tsv:md5,b0fa7435c4509396f474cf2152f9430d",
                            "subject_a.tumor.linx.vis_fusion.tsv:md5,7c3c61abb9279e277ba996bfaa98e102",
                            "subject_a.tumor.linx.vis_gene_exon.tsv:md5,2c1f2c84b93bf269ab4ddc4db385d752",
                            "subject_a.tumor.linx.vis_protein_domain.tsv:md5,1b1e08e5c9ee20583213be3728f0e1c6",
                            "subject_a.tumor.linx.vis_segments.tsv:md5,882f55efbc913f020ae0a7bf350a2a10",
                            "subject_a.tumor.linx.vis_sv_data.tsv:md5,a7319e3b9ea3df2da2a5acb121f2d9f2"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,5a0407da5642fd3fac02c26d740c0369"
                ],
                "annotation_dir": [
                    [
                        {
                            "id": "test",
                            "sample_id": "subject_a.tumor"
                        },
                        [
                            "linx.version:md5,a5fc97b5c4e6092096f0a1fdffd20159",
                            "subject_a.tumor.linx.breakend.tsv:md5,78e4cce3e07d1b0016d0a71b49db0e57",
                            "subject_a.tumor.linx.clusters.tsv:md5,49209060d580237991391282dbbea3af",
                            "subject_a.tumor.linx.driver.catalog.tsv:md5,1b18c47117b810430c3197f84c1db659",
                            "subject_a.tumor.linx.drivers.tsv:md5,8fb6e60a850979da67e3d3801c4a56d7",
                            "subject_a.tumor.linx.fusion.tsv:md5,cfaab2c97b2bddc258d2a5e866c55c97",
                            "subject_a.tumor.linx.links.tsv:md5,b1d1360fa6598420a8d9751ba2a4d4dc",
                            "subject_a.tumor.linx.neoepitope.tsv:md5,3c76a4e3521ccd5ac627857c2c5d17de",
                            "subject_a.tumor.linx.svs.tsv:md5,d6f1abd6f8678858e28eafe43fad4964",
                            "subject_a.tumor.linx.vis_copy_number.tsv:md5,b0fa7435c4509396f474cf2152f9430d",
                            "subject_a.tumor.linx.vis_fusion.tsv:md5,7c3c61abb9279e277ba996bfaa98e102",
                            "subject_a.tumor.linx.vis_gene_exon.tsv:md5,2c1f2c84b93bf269ab4ddc4db385d752",
                            "subject_a.tumor.linx.vis_protein_domain.tsv:md5,1b1e08e5c9ee20583213be3728f0e1c6",
                            "subject_a.tumor.linx.vis_segments.tsv:md5,882f55efbc913f020ae0a7bf350a2a10",
                            "subject_a.tumor.linx.vis_sv_data.tsv:md5,a7319e3b9ea3df2da2a5acb121f2d9f2"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5a0407da5642fd3fac02c26d740c0369"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-20T04:34:26.802584133"
    }
}